README.md

```diff
@@ -15,107 +16,92 @@ The *Perun Keystone Adapter* is a library written in Python that parses data pro
+```bash
 pip install git+https://github.com/deNBI/perunKeystoneAdapter@0.1.1
-~~~
+```
 
-## Example usage
+## Usage
 
 ### Commandline client
 
 The perun propagation service transfers a zipped tar file containing a users and groups file in scim format.
-The following script unzip and untar the propagated information and  add it to the keystone database. Keystone is addressed by environment variables (sourcing the openstack rc file) or  directly by passing an environemt map (not used in the example). The Openstack user needs at least permission to modifies Keystone
-
-```
-import sys
-import tarfile
-import os
-
-from denbi.bielefeld.perun.endpoint import Endpoint
-from denbi.bielefeld.perun.keystone import KeyStone
-
-if len (sys.argv) != 2:
-    print("usage: "+sys.argv[0]+" [perun_upload.tar.gz]")
-    sys.exit(1)
-
-fname = sys.argv[1]
-
-cwd = os.getcwd()
+The following script unzip and untar the propagated information and  add it to the keystone database. Keystone is addressed by environment variables (sourcing the openstack rc file) or  directly by passing an environemt map (not used in the example). The Openstack user needs at least permission to modify entries in Keystone
 
-# extract tar file
-tar = tarfile.open(fname, "r:gz")
-tar.extractall(path=cwd)
-tar.close()
-
-# import into keystone
-keystone = KeyStone(default_role="user",create_default_role=True, support_quotas= False, target_domain_name='elixir')
-endpoint = Endpoint(keystone=keystone,mode="denbi_portal_compute_center",support_quotas=False)
-endpoint.import_data(cwd+'/users.scim',cwd+'/groups.scim')
+```console
+$ perun_propagation perun_upload.tar.gz
 ```
 
 ### WSGI script
 
-The [example directory](python/example) also contains an [example](python/example/perun_propagation_service.py) for using the Perun Keystone Adapter library in a Python wsgi script. The script uses [flask](http://flask.pocoo.org/) to provide an upload function and run library functions in a separate thread. It can be simply tested starting the flask built-in webserver ...
+The python module also contains a built-in server version of the `perun_propagation` script. The script uses [flask](http://flask.pocoo.org/) to provide an upload function and run library functions in a separate thread. It can be simply tested starting the flask built-in webserver ...
 
-```
-$ python3 service.py
+```console
+$ perun_propagation_service
+ * Serving Flask app "denbi.scripts.perun_propagation_service" (lazy loading)
+ * Environment: production
+   WARNING: Do not use the development server in a production environment.
+   Use a production WSGI server instead.
+ * Debug mode: off
  * Running on http://127.0.0.1:5000/ (Press CTRL+C to quit)
-127.0.0.1 - - [23/Feb/2018 09:54:16] "PUT /upload HTTP/1.1" 200 -
-...
 ```
-... and using curl to upload sample propagation data.
 
-```
-$ curl -T file http://127.0.0.1:5000/upload
-```
+For running this in production it is typical to use something like `gunicorn` as follows:
```

Comment:
Well, actually in production it is recommended to use gunicorn **or** apache with a wsgi module. The built in server that you are running is just meant for development and that is why flask is printing the WARNING and recommends to use a wsgi based server such as apache wsgi: http://flask.pocoo.org/docs/0.12/deploying/mod_wsgi/

Reply:
I'll link to that documentation. But in general I prefer gunicorn since it gives you some more flexibility (apache, nginx, haproxy upstreams all work fine), is a working webserver out of the box (you don't have to point users to apache documentation), and is quite easy for users to get started with.